File: 75. Sort Colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
 var sortColors = function(arr) {
    let start = 0;
let mid = 0;
let end = arr.length - 1;

while (mid <= end) {
  if (arr[mid] == 0) {
    swap(arr, start, mid);
    start++;
    mid++;
  } else if (arr[mid] == 1) {
    mid++;
  } else if (arr[mid] == 2) {
    swap(arr, mid, end);
    end--;
  }
}
return arr;
};
const swap = (arr, i, j) => {
const temp = arr[i];
arr[i] = arr[j];
arr[j] = temp;
return arr;
};

File: 75. Sort Colors.java
 // One pass approach

 class Solution {
    public void sortColors(int[] nums) {
        int start=0;
        int mid=0;
        int end=nums.length-1;
         
        while(mid<=end){
            if(nums[mid]==0){
                swap(nums,start,mid);
                start++; mid++;
            }else if(nums[mid]==1){
                mid++;
            }
            else if(nums[mid]==2){
                swap(nums,mid,end);
                end--;
            }
             
        }
    }
    
   public void swap(int[] nums, int i, int j) {
	int temp = nums[i];
	nums[i] = nums[j];
	nums[j] = temp;
   }
}